tools/src/commands/IosNativeUnitTests.ts

```diff
@@ -0,0 +1,86 @@
+import spawnAsync from '@expo/spawn-async';
+import fs from 'fs-extra';
+import path from 'path';
+
+import * as Directories from '../Directories';
+import * as Packages from '../Packages';
+
+export async function iosNativeUnitTests() {
+  const packages = await Packages.getListOfPackagesAsync();
+  let packagesTested: string[] = [];
+  let errors: any[] = [];
+  for (const pkg of packages) {
+    if (
+      !pkg.isSupportedOnPlatform('ios') ||
+      !(await pkg.hasNativeTestsAsync('ios')) ||
+      !pkg.podspecName
+    ) {
+      continue;
+    }
+    packagesTested.push(pkg.packageName);
+
+    try {
+      await spawnAsync('fastlane', ['prepare_schemes', `pod:${pkg.podspecName}`], {
+        cwd: Directories.getExpoRepositoryRootDir(),
+        stdio: 'inherit',
+      });
+
+      // make schemes shared by moving them from xcodeproj/xcuserdata/runner.xcuserdatad/xcschemes
+      // to xcodeproj/xcshareddata/xcschemes
+      // otherwise they aren't visible to fastlane
+      const xcodeprojDir = path.join(
+        Directories.getIosDir(),
+        'Pods',
+        `${pkg.podspecName}.xcodeproj`
+      );
+      const destinationDir = path.join(xcodeprojDir, 'xcshareddata', 'xcschemes');
+      await fs.mkdirp(destinationDir);
+
+      // find user directory name, should be runner.xcuserdatad but depends on the OS username
+      const xcuserdataDirName = (await fs.readdir(path.join(xcodeprojDir, 'xcuserdata')))[0];
+
+      const xcschemesDir = path.join(xcodeprojDir, 'xcuserdata', xcuserdataDirName, 'xcschemes');
+      const xcschemesFiles = (await fs.readdir(xcschemesDir)).filter((file) =>
+        file.endsWith('.xcscheme')
+      );
+      if (!xcschemesFiles.length) {
+        throw new Error(`No scheme could be found to run tests for ${pkg.podspecName}`);
+      }
+      for (const file of xcschemesFiles) {
+        await fs.move(path.join(xcschemesDir, file), path.join(destinationDir, file), {
+          overwrite: true,
+        });
+      }
+
+      await spawnAsync('fastlane', ['test_module', `pod:${pkg.podspecName}`], {
+        cwd: Directories.getExpoRepositoryRootDir(),
+        stdio: 'inherit',
+      });
+    } catch (error) {
+      errors.push(error);
+    }
+  }
+  if (errors.length) {
+    console.error('One or more iOS unit tests failed:');
+    for (const error of errors) {
+      console.error(error.message);
+      console.error('stdout >', error.stdout);
+      console.error('stderr >', error.stderr);
+      if (error.message.startsWith('fastlane exited')) {
+        console.warn(
+          "Did you add unit tests to a package that didn't have unit tests before? If so, make sure to add the correct subspec to ios/Podfile."
+        );
+      }
```

Comment:
> If a package has tests but we want to skip running them, it would be convenient and intuitive to exclude them from the Podfile.

I think this exact scenario is very likely to happen as a result of an error / missed step, which is what I was hoping to head off with this error message. I don't think it's intuitive that you need to manually modify a project's Podfile in order to get a package's tests to run -- I could easily envision a scenario in which someone adds tests to expo-camera, sees that the unit test job passes, and doesn't realize that their new tests aren't actually running. (FWIW, on Android no additional configuration is necessary, and there is not currently a way to skip particular tests in the existing Android analog of this command/GHA.)

Would prefer to have some other way to purposely skip/exclude tests. Perhaps https://github.com/expo/expo/pull/12373 could be part of the solution?

Reply:
Maybe we can cross that bridge when we come to it. Thinking this through more, it's more likely someone would want to skip a single test case, not all of the tests for a package.